server/container/guice/guice-common/src/main/java/org/apache/james/utils/ConfigurationsPerformer.java

```diff
@@ -19,23 +19,42 @@
 
 package org.apache.james.utils;
 
+import java.util.List;
 import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
+import org.apache.james.lifecycle.api.Configurable;
+
+import com.github.fge.lambdas.Throwing;
 import com.google.inject.Inject;
 
+@SuppressWarnings("rawtypes")
 public class ConfigurationsPerformer {
 
     private final Set<ConfigurationPerformer> configurationPerformers;
+    private final List<Class<? extends Configurable>> configurables;
 
     @Inject
-    public ConfigurationsPerformer(Set<ConfigurationPerformer> configurationPerformers) throws Exception {
+    public ConfigurationsPerformer(Set<ConfigurationPerformer> configurationPerformers, Configurables configurables) {
         this.configurationPerformers = configurationPerformers;
+        this.configurables = configurables.get();
     }
 
     public void initModules() throws Exception {
-        for(ConfigurationPerformer configurationPerformer : configurationPerformers) {
-            configurationPerformer.initModule();
-        }
+        
+        Set<ConfigurationPerformer> processed = configurables.stream()
+            .flatMap(configurable -> configurationPerformerFor(configurable, configurationPerformers))
+            .peek(Throwing.consumer(x -> x.initModule()))
```

Comment:
you can extract the consumer into a local variable.
Or have initModule throw only runtime